Analysis/Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx

```diff
@@ -142,6 +173,228 @@ struct SelectTracks {
+  Configurable<double> d_minparamchange{"d_minparamchange", 1.e-3, "stop iterations if largest change of any X is smaller than this"};
+  Configurable<double> d_minrelchi2change{"d_minrelchi2change", 0.9, "stop iterations if chi2/chi2old > this"};
+  Configurable<bool> d_UseAbsDCA{"d_UseAbsDCA", true, "Use Abs DCAs"};
+
+  // quality cut
+  Configurable<bool> doCutQuality{"doCutQuality", true, "apply quality cuts"};
+
+  // track cuts for V0 daughters
+  Configurable<bool> tpcrefit{"tpcrefit", true, "request TPC refit V0 daughters"};
+  Configurable<int> mincrossedrows{"mincrossedrows", 50, "min crossed rows V0 daughters"};
+  Configurable<double> etamax{"etamax", 1.1, "max. pseudorapidity V0 daughters"};
+  Configurable<double> ptmin{"ptmin", 0.05, "min. pT V0 daughters"};
+
+  // bachelor cuts
+  //  Configurable<float> dcabachtopv{"dcabachtopv", .1, "DCA Bach To PV"};
+  //  Configurable<double> ptminbach{"ptminbach", -1., "min. track pT bachelor"};
+
+  // v0 cuts
+  Configurable<double> cospaV0{"cospaV0", .995, "CosPA V0"};                 // as in the task that create the V0s
+  Configurable<double> dcaXYnegtopv{"dcaXYnegtopv", .1, "DCA_XY Neg To PV"}; // check: in HF Run 2, it was 0 at filtering
+  Configurable<double> dcaXYpostopv{"dcaXYpostopv", .1, "DCA_XY Pos To PV"}; // check: in HF Run 2, it was 0 at filtering
+  Configurable<double> cutInvMassV0{"cutInvMassV0", 0.05, "V0 candidate invariant mass difference wrt PDG"};
+
+  // cascade cuts
+  Configurable<double> cutCascPtCandMin{"cutCascPtCandMin", -1., "min. pT of the 2-prong candidate"};              // PbPb 2018: use 1
+  Configurable<double> cutCascInvMassLc{"cutCascInvMassLc", 1., "Lc candidate invariant mass difference wrt PDG"}; // for PbPb 2018: use 0.2
+  //Configurable<double> cutCascDCADaughters{"cutCascDCADaughters", .1, "DCA between V0 and bachelor in cascade"};
+
+  // histograms
+  HistogramRegistry registry{
+    "registry",
+    {{"hvtx2_x", "2-prong candidates;#it{x}_{sec. vtx.} (cm);entries", {HistType::kTH1F, {{1000, -2., 2.}}}},
+     {"hvtx2_y", "2-prong candidates;#it{y}_{sec. vtx.} (cm);entries", {HistType::kTH1F, {{1000, -2., 2.}}}},
+     {"hvtx2_z", "2-prong candidates;#it{z}_{sec. vtx.} (cm);entries", {HistType::kTH1F, {{1000, -20., 20.}}}},
+     {"hmass2", "2-prong candidates;inv. mass (K0s p) (GeV/#it{c}^{2});entries", {HistType::kTH1F, {{500, 0., 5.}}}}}};
+
+  // NB: using FullTracks = soa::Join<Tracks, TracksCov, TracksExtra>; defined in Framework/Core/include/Framework/AnalysisDataModel.h
+  using MyTracks = soa::Join<aod::FullTracks, aod::HFSelTrack, aod::TracksExtended>;
+  //Partition<MyTracks> selectedTracks = aod::hf_seltrack::isSelProng >= 4;
+  // using SelectedV0s = soa::Filtered<aod::V0DataExt>;
+  double massP = RecoDecay::getMassPDG(kProton);
+  double massK0s = RecoDecay::getMassPDG(kK0Short);
+  double massPi = RecoDecay::getMassPDG(kPiPlus);
+  double massLc = RecoDecay::getMassPDG(4122);
```

Comment:
`const`?

Reply:
done

Reply:
Can't be const actually:
```
/home/zampolli/SOFT/alibuild/ali-o2-analysis/sw/SOURCES/O2/dev/0/Analysis/Tasks/PWGHF/HFCandidateCreatorCascade.cxx: In member function ‘void HFCandidateCreatorCascade::process(const Collisions&, const HfTrackIndexCasc&, const BigTracks&, const V0DataExt&)’:
/home/zampolli/SOFT/alibuild/ali-o2-analysis/sw/SOURCES/O2/dev/0/Analysis/Tasks/PWGHF/HFCandidateCreatorCascade.cxx:145:19: error: assignment of read-only member ‘HFCandidateCreatorCascade::mass2K0sP’
  145 |         mass2K0sP = RecoDecay::M(arrayMomenta, array{massK0s, massP});
```
(from the other class, but it is the same here)

Reply:
My comment still applies.

Reply:
I don't understand the comment. I cannot make them `const`.